bus_analyzer/bus_analyzer/post_processing.py

```diff
@@ -0,0 +1,144 @@
+import itertools
+import math
+import operator
+
+MAX_SPEED = 45 / 3600
+
+
+def post_process_miles(seconds, miles, days):
+	miles = [math.nan if mile is None else mile for mile in miles]
+	good = []
+	suspect = []
+	for i in range(1, len(seconds) - 1):
+		if math.isnan(miles[i]) or miles[i] <= 100:
+			suspect.append(i)
+			continue
+		if days[i] is None or days[i] == 'score':
+			suspect.append(i)
+			continue
+		previous_diff = miles[i] - miles[i - 1]
+		if previous_diff < 0 or previous_diff > MAX_SPEED * (seconds[i] - seconds[i - 1]):
+			suspect.append(i)
+			continue
+		next_diff = miles[i + 1] - miles[i]
+		if next_diff < 0 or next_diff > MAX_SPEED * (seconds[i + 1] - seconds[i]):
+			suspect.append(i)
+			continue
+		# handle big jumps to apparently good data
+		if good and miles[i] - miles[good[-1]] > MAX_SPEED * (seconds[i] - seconds[good[-1]]):
+			suspect.append(i)
+			continue
+		# try to filter out bad data at the start
+		if not good and miles[i] > 1000:
```

Comment:
won't this prevent us from ever marking anything good when processing data that starts past 1000 miles (ie. halfway through point 3)?